tests/unit/preprocessor/_time/test_time.py

```diff
@@ -171,6 +171,12 @@ def test_extract_time_no_time(self):
         sliced = extract_time(cube, 1950, 1, 1, 1950, 12, 31)
         assert cube == sliced
 
+    def test_extract_time_none_year(self):
+        """Test extract_time when one of the years is None."""
+        cube = self.cube.coord('time').guess_bounds()
+        with assert_raises(ValueError):
+            extract_time(cube, None, 1, 1, 1950, 2, 1)
```

Comment:
this covers one of two cases: you'd ideally need a (None, None) case as well

Reply:
also, it's very useful to check for the correct exception raised, ie check for its string `str(err.value)` if you call it with `with pytest.raises... as err:`